### PR TITLE
test(api): Added unit tests 

### PR DESCRIPTION
**Pull request by kvol**
Added unit test for  connector_resync_table, connector_delete_mock, connector_setup_tests

### CODE REVIEW

File: tests/connector_delete_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		})
+
+	service := ftClient.NewConnectorDelete().ConnectorID("connector_id")
+
+	// act
+	response, err := service.Do(context.Background())
+
+	// assert
+	if err != nil {
+		assertConnectorDeleteResponse(t, response, "NotFound_Integration", "Cannot find entity 'Integration' with id")
```
Comment (LukaDevic) [inline tests/connector_delete_mock_test.go:56]: Update response message to  "Cannot find entity 'Integration' with id 'connector_id'"

File: tests/connector_delete_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		t.Logf("%+v\n", response)
+		t.Error(err)
+	}
+
+}
+
+func prepareConnectorDeleteResponse(code string, message string) string {
+	var s = "{\"code\": \"" + code + "\" , \"message\": \"" + message + "\"}"
+	return s
+
```
Comment (LukaDevic) [inline tests/connector_delete_mock_test.go:68]: Please delete empty line

File: tests/connector_resync_table_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	// assert
+	interactions := mockClient.Interactions()
+	assertEqual(t, len(interactions), 1)
+	assertEqual(t, interactions[0].Handler, handler)
+	assertEqual(t, handler.Interactions, 1)
+
+	assertConnectorReSyncTableResponse(t, response, "BadRequest", "Invalid request")
+}
+
+func TestConnectorReSyncTable_WithNilConnectorID(t *testing.T) {
```
Comment (LukaDevic) [inline tests/connector_resync_table_mock_test.go:43]: Please remove '_' from test name

File: tests/connector_resync_table_mock_test.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+	// assert
+	interactions := mockClient.Interactions()
+	assertEqual(t, len(interactions), 0)
+	assertEqual(t, response, fivetran.ConnectorReSyncTableResponse{})
+	assertIsNotNil(t, err)
+	assertEqual(t, err.Error(), "missing required ConnectorID")
+}
+
+func TestConnectorReSyncTable_WithNilSchema(t *testing.T) {
```
Comment (LukaDevic) [inline tests/connector_resync_table_mock_test.go:60]: Please remove '_' from test name

File: tests/connector_resync_table_mock_test.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+	// assert
+	interactions := mockClient.Interactions()
+	assertEqual(t, len(interactions), 0)
+	assertEqual(t, response, fivetran.ConnectorReSyncTableResponse{})
+	assertIsNotNil(t, err)
+	assertEqual(t, err.Error(), "missing required Schema")
+}
+
+func TestConnectorReSyncTable_WithNilTable(t *testing.T) {
```
Comment (LukaDevic) [inline tests/connector_resync_table_mock_test.go:77]: Please remove '_' from test name

File: tests/connector_setup_tests_test.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
+package tests
```
Comment (LukaDevic) [inline tests/connector_setup_tests_test.go:1]: Please remove '_tests' from this file name.

File: tests/connector_setup_tests_test.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/fivetran/go-fivetran"
+	"github.com/fivetran/go-fivetran/tests/mock"
+)
+
+func TestConnectorSetupTests_Do(t *testing.T) {
```
Comment (LukaDevic) [inline tests/connector_setup_tests_test.go:12]: Please remove '_' from test name.

File: tests/connector_setup_tests_test.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+		t.Logf("%+v\n", response)
+		t.Error(err)
+	}
+
+	interactions := mockClient.Interactions()
+	assertEqual(t, len(interactions), 1)
+	assertEqual(t, interactions[0].Handler, handler)
+	assertEqual(t, handler.Interactions, 1)
+
+	// assert
```
Comment (LukaDevic) [inline tests/connector_setup_tests_test.go:35]: Move '// assert' up to line 29

File: tests/connector_setup_tests_test.go [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+	assertNotEmpty(t, response.Data.Status.UpdateState)
+
+	// Assert setup tests
+	for _, test := range response.Data.SetupTests {
+		assertNotEmpty(t, test.Title)
+		assertNotEmpty(t, test.Status)
+		assertNotEmpty(t, test.Message)
+	}
+
+	// Add assertions for specific fields in the ConnectorConfigResponse struct
```
Comment (LukaDevic) [inline tests/connector_setup_tests_test.go:87]: Please add assertions for specific fields in the ConnectorConfigResponse struct

**Review by LukaDevic — CHANGES_REQUESTED** [review]

**Comment by LukaDevic** [comment]
Hi @kvol, please also fix 3 unresolved issues mentioned in comments. 
Also tests are not passing, asserted msg needs to be updated.

File: tests/connector_setup_tests_test.go (hunk 6 above)
Comment (kvol) [reply tests/connector_setup_tests_test.go:1]: this test for connector_setup_tests, i fink not correct violate the general naming rules

**Review by beevital — APPROVED** [review]
LGTM.

**Comment by beevital** [comment]
> Also tests are not passing, asserted msg needs to be updated.

@LukaDevic the problem is on API side, there was no changes related to these failing tests here.

File: tests/connector_setup_tests_test.go (hunk 6 above)
Comment (beevital) [reply tests/connector_setup_tests_test.go:1]: this is correct naming as original source file named `connector_setup_tests.go`, so this is correct naming.

**Review by LukaDevic — APPROVED** [review]
App

**Pull request merged**